domain/storageprovisioning/state/types.go

```diff
@@ -30,6 +30,23 @@ func (l attachmentLives) Iter(yield func(string, life.Life) bool) {
 	}
 }
 
+type storageAttachmentLife struct {
+	ID   string    `db:"storage_id"`
```

Comment:
Can you change `ID` to `StorageID` so it is explicit what is being referred to or even `StorageInstanceID`